### DISCUSSION TITLE
abacus with gcc-12

### DISCUSSION THREAD

**Discussion by QuantumMisaka** [deepmodeling/abacus-develop#2576] [Ideas]
Has anyone tested the performance of abacus compiled by gcc-12.2.0 ?
my server has only gcc-8.5.0 and gcc-12.2.0, and the I tried gcc-12.2.0 to do gcc-toolchain compilation
From a single cell-relax test, abacus by gcc-12.2.0 show a better performance 
but in document, abacus support cxx-11 only ?

**Comment by Satinelamp** [comment]
Is there a comparison of the time of each electronic step?
Maybe you could post your test case, I could try it.

**Reply by QuantumMisaka** [reply]
[Fe5C2-CHCOH.tar.gz](https://github.com/deepmodeling/abacus-develop/files/11615427/Fe5C2-CHCOH.tar.gz)

This is my test case
when I do calculation with 64core (in 1 node) and set `OMP_NUM_THREADS=4`, I noticed that time of each electronic step was about 36-40 s in my ABACUS compiled by gcc-10.2.0 before , while I use gcc-12.2.0 to re-compile abacus (and all other dependencies), I noticed that time of each electronic step is about 30-33 s, it seems to be speeded up.

**Comment by caic99** [comment]
@QuantumMisaka Please note that ABACUS requires the c++ language standard to be c++11, while not requiring a specified gcc version. Since gcc-12 does support c++11, this should work as expected.
Generally, newer compiler has better optimizations overall. And the compilation flags on optimizing also effects on the speed.

**Reply by QuantumMisaka** [reply]
Good explanation, Thanks!
